File: tiq-e/others/5/main.go
package main

import (
	"fmt"
	"time"
)

func isValid(s string) bool {
	stack := make([]rune, 0)
	match := map[rune]rune{
		')': '(',
		']': '[',
		'}': '{',
	}
	for _, char := range s {
		switch char {
		case '(', '{', '[':
			stack = append(stack, char)
		case ')', '}', ']':
			if len(stack) == 0 || stack[len(stack)-1] != match[char] {
				return false
			}
			stack = stack[:len(stack)-1]
		}
	}

	return len(stack) == 0
}

func main() {
	x := "()[]{}"

	start := time.Now()
	result := isValid(x)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/others/5/main.ts
import {performance} from "perf_hooks";

function isValid(s: string): boolean {
    let stack = [];
    for (const char of s) {
        switch (char) {
            case "(":
                stack.push(")");
                break;
            case "[":
                stack.push("]");
                break;
            case "{":
                stack.push("}");
                break;
            default:
                if (char !== stack.pop()) {
                    return false;
                }
        }
    }
    return !stack.length;
}

const x = "()[]{}";

const start = performance.now();
const result = isValid(x);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);
